### PR TITLE
Expose get_queue method for context in API

### PR DESCRIPTION
**Pull request by umangyadav**

### CODE REVIEW

File: src/api/include/migraphx/migraphx.hpp [hunk 1]
```diff
@@ -777,6 +777,13 @@ struct context
     migraphx_context_t ctx;
 
     void finish() const { call(&migraphx_context_finish, ctx); }
+
+    void* get_queue()
```
Comment (pfultz2) [inline src/api/include/migraphx/migraphx.hpp:781]: This should do the cast so we can type check this in the future:

```cpp
    template<class T>
    T get_queue() const
    {
        void* out;
        call(&migraphx_context_get_queue, &out, ctx);
        // TODO: Check type
        return reinterpret_cast<T>(out);
    }
```

File: test/api/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,12 @@
-
 function(add_api_test TEST_NAME TEST_SRC TEST_DIR)
     set(NAME test_api_${TEST_NAME})
     add_executable(${NAME} EXCLUDE_FROM_ALL ${TEST_SRC})
     rocm_clang_tidy_check(${NAME})
     target_link_libraries(${NAME} migraphx_c migraphx)
     target_include_directories(${NAME} PUBLIC ../include)
+    if(MIGRAPHX_ENABLE_GPU)
+    target_link_libraries(${NAME} migraphx_gpu)
```
Comment (pfultz2) [inline test/api/CMakeLists.txt:8]: This should be done outside of this function.

File: src/api/include/migraphx/migraphx.hpp (hunk 1 above)
Comment (umangyadav) [reply src/api/include/migraphx/migraphx.hpp:781]: made changes.

File: test/api/CMakeLists.txt (hunk 2 above)
Comment (umangyadav) [reply test/api/CMakeLists.txt]: Done.

File: src/api/include/migraphx/migraphx.hpp (hunk 1 above)
Comment (pfultz2) [reply src/api/include/migraphx/migraphx.hpp:781]: It needs to return `T` not `void*`.

Comment (umangyadav) [reply src/api/include/migraphx/migraphx.hpp]: Done.

**Review by kahmed10 — APPROVED** [review]

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**